### PR TITLE
Handle EKS add-ons with native Terraform instead of kubectl

### PR DESCRIPTION
**Pull request by abstrask**
Replace null resources using `kubectl` with native [`aws_eks_addon`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_addon) resources.

Has the added benefit that the CNI will now be pulled from the ECR registry in the same region as cluster, instead of "us-west-2".

Minimum supported Kubernetes cluster version as of this change is 1.18.

### CODE REVIEW

**Comment by abstrask** [comment]
@avnes Can you test if this change updates the CoreDNS cluster role, when upgrading to the latest CoreDNS version? Or help me test it out?

**Comment by abstrask** [comment]
Test plan:

- New 1.19 cluster using same modules as current EKS pipeline (?ref=0.2.117)
- Output 'system:coredns' cluster role (look for 'endpointslices')
- Apply current branch on cluster phase (native TF addon management)
- Output 'system:coredns' cluster role (look for 'endpointslices')
- Upgrade cluster to 1.20 also using current branch
- Output 'system:coredns' cluster role (look for 'endpointslices')

**Comment by abstrask** [comment]
Cluster role gets amended accordingly when upgrading to cluster 1.20, so I guess this can replace PR #232, @avnes ?

File: _sub/compute/eks-addons/dependencies.tf [hunk 1]
```diff
@@ -4,42 +4,32 @@
...
   vpccni_version_map = {
-    "1.15" = "1.7.5"
-    "1.16" = "1.7.5"
-    "1.17" = "1.7.5"
-    "1.18" = "1.7.5"
-    "1.19" = "1.7.5"
-    "1.20" = ""
+    "1.18" = "v1.7.5-eksbuild.2"
+    "1.19" = "v1.7.5-eksbuild.2"
+    "1.20" = "v1.7.5-eksbuild.2"
```
Comment (avnes) [inline _sub/compute/eks-addons/dependencies.tf:14]: https://docs.aws.amazon.com/eks/latest/userguide/managing-vpc-cni.html#updating-vpc-cni-add-on recommends using CNI version 1.7.10

Comment (avnes) [reply _sub/compute/eks-addons/dependencies.tf:14]: It is also CNI version 1.7.10 tested with in the null_provider update

**Review by avnes — CHANGES_REQUESTED** [review]
I can confirm this works really well, and gets the job done when upgrading from EKS 1.19 and EKS 1.20 - I just have one doubt about the CNI version used. Please see other comment.

**Comment by abstrask** [comment]
I get this using AWS CLI:

```
$ aws eks describe-addon-versions --kubernetes-version 1.18 --addon-name vpc-cni | jq -r '.addons[].addonVersions[] | select(.compatibilities[].defaultVersion == true) | .addonVersion'
v1.7.5-eksbuild.2

$ aws eks describe-addon-versions --kubernetes-version 1.19 --addon-name vpc-cni | jq -r '.addons[].addonVersions[] | select(.compatibilities[].defaultVersion == true) | .addonVersion'
v1.7.5-eksbuild.2

$ aws eks describe-addon-versions --kubernetes-version 1.20 --addon-name vpc-cni | jq -r '.addons[].addonVersions[] | select(.compatibilities[].defaultVersion == true) | .addonVersion'
v1.7.5-eksbuild.2
```

**Review by avnes — APPROVED** [review]

**Pull request merged**